### DISCUSSION TITLE
Subscribe() and getState() examples broken in v4.3.3

### DISCUSSION THREAD

**Discussion by Sp1d3rbite** [pmndrs/zustand#1646] [Q&A]
Previously I was using version 3.6.5 and successfully used the subscribe and getState functions.
After upgrading to (specifically) version 4.3.3 both of those broke so I referred to the docs and the migration guide.

As far as I can tell no changes needed to be made on my side so I tried the doc example and saw those failed as well.
Any thoughts on why this is, or am I missing something?

![rider64_mrVXDT6JPz](https://user-images.githubusercontent.com/29540148/220080009-573f981a-b7fd-4bb4-ac07-67039fcf9481.png)

![rider64_fEWFSuZg1H](https://user-images.githubusercontent.com/29540148/220080138-41f12692-a7ba-4cc4-b495-59d947031755.png)

**Comment by dai-shi** [comment]
Hi, thanks for reporting.
Is the example from README? It's not broken. It's just in JS.

> ⚠️ This readme is written for JavaScript users. If you are a TypeScript user, don't miss [TypeScript Usage](https://github.com/pmndrs/zustand#typescript-usage).


It's under debate: #1430

**Reply by Sp1d3rbite** [reply]
Hi, thanks for the reply!
My entire project and the script showing the errors are fully written in Js.
I had just copied the code directly from the Readme docs trying to sandbox the errors happening.

**Reply by dai-shi** [reply]
I just copied the code from readme, and it seems working.
https://codesandbox.io/s/trusting-water-jq44jw?file=/src/App.js

**Reply by Sp1d3rbite** [reply]
Hmm, you are right. I thought it was a problem with me using Vite instead of CRA but trying that and it also seems to work correctly there: https://codesandbox.io/s/boring-hermann-98ztk1?file=/src/App.jsx

But I am still confused about the problem. When using v3.6.5 everything is correct, see below:
![rider64_poriTydIJY](https://user-images.githubusercontent.com/29540148/220354357-00607355-68ad-43dd-a004-c7e74616aa70.png)

But when I upgrade to zustand v4.3.3 it still gives me this error and, more importantly, fails code execution:
![rider64_MlI1fNINeN](https://user-images.githubusercontent.com/29540148/220354498-f0e18267-37b2-4be3-83b2-4e0fccfe9a09.png)

The only difference between the two is the version of Zustand used.

**Reply by dai-shi** [reply]
Yes, and that's how we _improved_ TypeScript typing. It's breaking change. If you are using TS and middleware, you should use `create()(...)` instead of `create(...)`. Please read the typescript guide for details.
